dima/react-movies/src/pages/moviepage/moviepage.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useEffect } from "react";
+import { useHistory, useLocation, useParams } from "react-router";
+import ContentContainer from "../../components/contentContainer";
+import CurrentMovieGenreFilter from "../../components/currentMovieGenreFilter";
+import ErrorBoundary from "../../components/errorBoundary";
+import Footer from "../../components/footer";
+import Header from "../../components/header";
+import MoviePresent from "../../components/moviePresent/moviePresent";
+import MoviesContainer from "../../components/moviesContainer";
+import QueryString from "query-string";
+
+import "./moviepage.css";
+import { useDispatch, useSelector } from "react-redux";
+import { fetchMovie, resetMovie } from "../../redux/moviesReducer";
+import { StateInterface } from "../../interfaces/stateInterface";
+import MovieInterface from "../../interfaces/movieInterface";
+import MoviesDataInterface from "../../interfaces/moviesDataInterface";
+
+const MoviePage = () => {
+  let location = useLocation();
+  let params = useParams<{ id: string }>();
+  let history = useHistory();
+  const dispatch = useDispatch();
+  const movie = useSelector<StateInterface, MovieInterface | null>(
+    (state) => state.movie
+  );
+  const movies = useSelector<StateInterface, MoviesDataInterface>(
+    (state) => state.movies
+  );
+  const loadingMovies = useSelector<StateInterface, boolean>(
+    (state) => state.loadingMovies
+  );
+  const loadingMovie = useSelector<StateInterface, boolean>(
+    (state) => state.loadingMovie
+  );
+
+  useEffect(() => {
+    const fetchAllMovies = (pageNum: number): void => {
+      dispatch(fetchMovie({ id: params.id, offset: pageNum * 9 }));
+    };
+    const checkPageNumbersAndFecthMovie = (): void => {
+      const urlParams = QueryString.parse(location.search);
+      let pageNum = urlParams.page ? Number(urlParams.page.toString()) - 1 : 0;
+      fetchAllMovies(pageNum);
```

Comment:
It's a movie page, but I don't see how you pass a movie ID  param to make a movie call?